`test_refreshable_mat_view_replicated` is flaky

https://play.clickhouse.com/play?user=play#U0VMRUNUIHB1bGxfcmVxdWVzdF9udW1iZXIgYXMgcHIsIGNoZWNrX3N0YXJ0X3RpbWUsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgdGVzdF9zdGF0dXMsIHJlcG9ydF91cmwKRlJPTSBjaGVja3MKV0hFUkUgMQogICAgLS0gQU5EIHB1bGxfcmVxdWVzdF9udW1iZXIgPSAwCiAgICBBTkQgdGVzdF9zdGF0dXMgPSAnRVJST1InCiAgICBBTkQgdG9EYXRlKGNoZWNrX3N0YXJ0X3RpbWUpID4gJzIwMjQtMTItMDEnCiAgICBBTkQgdGVzdF9uYW1lIGxpa2UgJyV0ZXN0X3JlZnJlc2hhYmxlX21hdF92aWV3X3JlcGxpY2F0ZWQlJwpPUkRFUiBCWSBjaGVja19zdGFydF90aW1lIERFU0M=

```
==================================== ERRORS ====================================
______________________ ERROR at setup of test_query_fail _______________________
[gw0] linux -- Python 3.10.12 /usr/bin/python3

    @pytest.fixture(scope="function")
    def fn3_setup_tables():
        node.query("DROP TABLE IF EXISTS tgt1 ON CLUSTER default")
        node.query("DROP TABLE IF EXISTS test_rmv ON CLUSTER default")
        node.query("DROP TABLE IF EXISTS test_db.test_rmv")
    
>       node.query(
            f"CREATE TABLE tgt1 ON CLUSTER default (a DateTime) ENGINE = ReplicatedMergeTree ORDER BY tuple()"
        )

test_refreshable_mat_view_replicated/test.py:562: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
helpers/cluster.py:3677: in query
    return self.client.query(
helpers/client.py:39: in wrap
    return func(self, *args, **kwargs)
helpers/client.py:79: in query
    ).get_answer()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <helpers.client.CommandRequest object at 0xff5306a74e20>

    def get_answer(self):
        self.process.wait(timeout=DEFAULT_QUERY_TIMEOUT)
        self.stdout_file.seek(0)
        self.stderr_file.seek(0)
    
        stdout = self.stdout_file.read().decode("utf-8", errors="replace")
        stderr = self.stderr_file.read().decode("utf-8", errors="replace")
    
        if (
            self.timer is not None
            and not self.process_finished_before_timeout
            and not self.ignore_error
        ):
            logging.debug(f"Timed out. Last stdout:{stdout}, stderr:{stderr}")
            raise QueryTimeoutExceedException("Client timed out!")
    
        if (
            self.process.returncode != 0 or self.remove_trash_from_stderr(stderr)
        ) and not self.ignore_error:
>           raise QueryRuntimeException(
                "Client failed! Return code: {}, stderr: {}".format(
                    self.process.returncode, stderr
                ),
                self.process.returncode,
                stderr,
            )
E           helpers.client.QueryRuntimeException: Client failed! Return code: 57, stderr: Received exception from server (version 25.1.1):
E           Code: 57. DB::Exception: Received from 172.16.2.5:9000. DB::Exception: Table default.tgt1 already exists. Stack trace:
E           
E           0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c121b04
E           1. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x0000000007d45e3c
E           2. DB::Exception::Exception<String, String>(int, FormatStringHelperImpl<std::type_identity<String>::type, std::type_identity<String>::type>, String&&, String&&) @ 0x0000000007d47adc
E           3. DB::InterpreterCreateQuery::doCreateTable(DB::ASTCreateQuery&, DB::InterpreterCreateQuery::TableProperties const&, std::unique_ptr<DB::DDLGuard, std::default_delete<DB::DDLGuard>>&, DB::LoadingStrictnessLevel) @ 0x000000001027f3f8
E           4. DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) @ 0x00000000102761a4
E           5. DB::InterpreterCreateQuery::execute() @ 0x0000000010283b20
E           6. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x00000000107b23d4
E           7. DB::executeQuery(DB::ReadBuffer&, DB::WriteBuffer&, bool, std::shared_ptr<DB::Context>, std::function<void (DB::QueryResultDetails const&)>, DB::QueryFlags, std::optional<DB::FormatSettings> const&, std::function<void (DB::IOutputFormat&, String const&, std::shared_ptr<DB::Context const> const&, std::optional<DB::FormatSettings> const&)>) @ 0x00000000107b6918
E           8. DB::DDLWorker::tryExecuteQuery(DB::DDLTaskBase&, std::shared_ptr<zkutil::ZooKeeper> const&) @ 0x000000000ff80c90
E           9. DB::DDLWorker::processTask(DB::DDLTaskBase&, std::shared_ptr<zkutil::ZooKeeper> const&) @ 0x000000000ff7f420
E           10. DB::DatabaseReplicatedDDLWorker::tryEnqueueAndExecuteEntry(DB::DDLLogEntry&, std::shared_ptr<DB::Context const>) @ 0x000000000fab2e20
E           11. DB::DatabaseReplicated::tryEnqueueReplicatedDDL(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context const>, DB::QueryFlags) @ 0x000000000fa8c85c
E           12. DB::InterpreterCreateQuery::createTable(DB::ASTCreateQuery&) @ 0x0000000010276000
E           13. DB::InterpreterCreateQuery::execute() @ 0x0000000010283b20
E           14. DB::executeQueryImpl(char const*, char const*, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x00000000107b23d4
E           15. DB::executeQuery(String const&, std::shared_ptr<DB::Context>, DB::QueryFlags, DB::QueryProcessingStage::Enum) @ 0x00000000107aeacc
E           16. DB::TCPHandler::runImpl() @ 0x0000000011830778
E           17. DB::TCPHandler::run() @ 0x0000000011847f44
E           18. Poco::Net::TCPServerConnection::start() @ 0x000000001412ab58
E           19. Poco::Net::TCPServerDispatcher::run() @ 0x000000001412b034
E           20. Poco::PooledThread::run() @ 0x00000000140f4858
E           21. Poco::ThreadImpl::runnableEntry(void*) @ 0x00000000140f2c90
E           22. ? @ 0x000000000007d5c8
E           23. ? @ 0x00000000000e5edc
E           . (TABLE_ALREADY_EXISTS)
E           (query: CREATE TABLE tgt1 ON CLUSTER default (a DateTime) ENGINE = ReplicatedMergeTree ORDER BY tuple())

helpers/client.py:248: QueryRuntimeException
---------------------------- Captured stderr setup -----------------------------
Executing query DROP TABLE IF EXISTS tgt1 ON CLUSTER default on node1_1
Executing query DROP TABLE IF EXISTS test_rmv ON CLUSTER default on node1_1
Executing query DROP TABLE IF EXISTS test_db.test_rmv on node1_1
Executing query CREATE TABLE tgt1 ON CLUSTER default (a DateTime) ENGINE = ReplicatedMergeTree ORDER BY tuple() on node1_1
------------------------------ Captured log setup ------------------------------
2024-12-16 23:28:15 [ 677 ] DEBUG : Executing query DROP TABLE IF EXISTS tgt1 ON CLUSTER default on node1_1 (cluster.py:3676, query)
2024-12-16 23:28:15 [ 677 ] DEBUG : Executing query DROP TABLE IF EXISTS test_rmv ON CLUSTER default on node1_1 (cluster.py:3676, query)
2024-12-16 23:28:16 [ 677 ] DEBUG : Executing query DROP TABLE IF EXISTS test_db.test_rmv on node1_1 (cluster.py:3676, query)
2024-12-16 23:28:16 [ 677 ] DEBUG : Executing query CREATE TABLE tgt1 ON CLUSTER default (a DateTime) ENGINE = ReplicatedMergeTree ORDER BY tuple() on node1_1 (cluster.py:3676, query)
```

--- a/tests/integration/test_refreshable_mat_view_replicated/test.py
+++ b/tests/integration/test_refreshable_mat_view_replicated/test.py
@@ -217,6 +217,11 @@ def fn_setup_tables():
     )
     node.query(f"INSERT INTO src1 VALUES ('2020-01-01', 1), ('2020-01-02', 2)")
 
+    yield
+
+    node.query("DROP TABLE IF EXISTS test_rmv ON CLUSTER default")
+    node.query("DROP TABLE IF EXISTS test_db.test_rmv")
+
 
 @pytest.mark.parametrize(
     "select_query",
@@ -280,7 +285,7 @@ def test_append(
 @pytest.mark.parametrize("if_not_exists", [True, False])
 @pytest.mark.parametrize("depends_on", [None, ["default.dummy_rmv"]])
 @pytest.mark.parametrize("empty", [True, False])
-@pytest.mark.parametrize("database_name", ["test_db"])  # None,
+@pytest.mark.parametrize("database_name", ["test_db"])
 @pytest.mark.parametrize(
     "settings",
     [
@@ -555,8 +560,8 @@ def test_long_query_cancel(fn_setup_tables):
 
 @pytest.fixture(scope="function")
 def fn3_setup_tables():
-    node.query("DROP TABLE IF EXISTS tgt1 ON CLUSTER default")
-    node.query("DROP TABLE IF EXISTS test_rmv ON CLUSTER default")
+    node.query("DROP TABLE IF EXISTS tgt1 ON CLUSTER default SYNC")
+    node.query("DROP TABLE IF EXISTS test_rmv ON CLUSTER default SYNC")
     node.query("DROP TABLE IF EXISTS test_db.test_rmv")
 
     node.query(

